src/util/address.rs

```diff
@@ -511,13 +513,45 @@ impl Address {
         }
     }
 
-    /// Create a pay to taproot address
-    pub fn p2tr(taptweaked_key: schnorrsig::PublicKey, network: Network) -> Address {
+    // /// Create a pay to taproot address
+    // pub fn p2tr(taptweaked_key: schnorrsig::PublicKey, network: Network) -> Address {
+    //     Address {
+    //         network: network,
+    //         payload: Payload::WitnessProgram {
+    //             version: WitnessVersion::V1,
+    //             program: taptweaked_key.serialize().to_vec()
+    //         }
+    //     }
+    // }
+
+    /// Create a pay to taproot address from untweaked key
+    pub fn p2tr<C: Verification>(
+        secp: Secp256k1<C>, 
+        internal_key: UntweakedPublicKey, 
+        merkle_root: Option<TapBranchHash>, 
+        network: Network
+    ) -> Address {
         Address {
             network: network,
             payload: Payload::WitnessProgram {
                 version: WitnessVersion::V1,
-                program: taptweaked_key.serialize().to_vec()
+                program: internal_key.tweak(secp, merkle_root).into_raw().serialize().to_vec()
+            }
+        }
+    }
+
+    /// Create a pay to taproot address from a tweaked key.
+    /// 
+    /// This method is dangerous and can lead to loss of funds if used incorrectly.
+    pub fn dangerous_p2tr<C: Verification>(
+        internal_key: TweakedPublicKey, 
```

Comment:
According to BIP-341 the tweaked key is named differently
```suggestion
        output_key: TweakedPublicKey, 
```